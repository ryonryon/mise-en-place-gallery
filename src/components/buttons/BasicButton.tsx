import styled from "@emotion/styled";
import * as React from "react";

import { secondary, white } from "../../constants/colors";

const BasicStyledButton = styled.button`
  background-color: ${white};
  border-radius: 5px;
  border: none;
  color: ${secondary};
  height: 30px;
  width: 150px;
  cursor: pointer;
  margin: 16px;
  &:hover {
    background-color: rgba(221, 221, 221, 0.5);
  }
`;

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {}

const BasicButton = ({ ...rest }: Props) => <BasicStyledButton {...rest}></BasicStyledButton>;

export default BasicButton;
