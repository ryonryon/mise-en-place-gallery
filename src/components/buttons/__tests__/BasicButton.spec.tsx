import * as React from "react";
import * as renderer from "react-test-renderer";

import BasicButton from "../BasicButton";

describe("BasicButton test", () => {
  it("tests BasicButton renders correctly ", () => {
    const basicButton = renderer.create(<BasicButton />).toJSON();
    expect(basicButton).toMatchSnapshot();
  });
});
