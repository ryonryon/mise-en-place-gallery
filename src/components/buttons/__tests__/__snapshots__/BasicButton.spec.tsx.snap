// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicButton test tests BasicButton renders correctly  1`] = `
<button
  className="css-1qq2dj4"
/>
`;
